src/main/java/com/arpnetworking/metrics/common/parsers/exceptions/ParsingException.java

```diff
@@ -21,25 +21,34 @@
  * @author Brandon Arp (brandonarp at gmail dot com)
  */
 public class ParsingException extends Exception {
-
     /**
      * Public constructor with a description.
      *
      * @param message Describes the exceptional condition.
+     * @param offendingData The raw data that failed to parse.
      */
-    public ParsingException(final String message) {
+    public ParsingException(final String message, final byte[] offendingData) {
         super(message);
+        _offendingData = offendingData.clone();
```

Comment:
This is really expensive. Is this worth it? 

Alternatively the exception could accept a list and we could do Arrays.asList() and the wrap it as with unmodifiable list in the caller. Zero copy and anyone with access to the exception member still can't modify it. Here we expose the exception member for write access via its getter. 

Moral of the stories arrays are always evil. We should probably do more work with Jackson and streams.

-- also if you expecting it to get logged you should include @Loggable on the exception class.